diff --git a/dist/index.js b/dist/index.js
index 0bf7329fc3b42d1f4880ce433f9f950a2b3f6f42..8a295d7e81daf5957883da49f258c219bc88114f 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -171,10 +171,10 @@ function ViteSentry(options) {
         */
         configResolved(config) {
             pluginState.sourcemapsCreated = !!config.build.sourcemap;
-            pluginState.isProduction = config.isProduction;
+            pluginState.isProduction = config.mode === "production";
             pluginState.enabled =
                 pluginState.sourcemapsCreated &&
-                    (skipEnvironmentCheck || config.isProduction);
+                    (skipEnvironmentCheck || pluginState.isProduction);
         },
         /*
           Resolve id for virtual module
diff --git a/dist/index.mjs b/dist/index.mjs
index 9edaed02fbfde8f2920b9dddaf7222e0c6c01920..2bf0b1dc21b26e400180424802536d7fd74ded86 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -169,10 +169,10 @@ function ViteSentry(options) {
         */
         configResolved(config) {
             pluginState.sourcemapsCreated = !!config.build.sourcemap;
-            pluginState.isProduction = config.isProduction;
+            pluginState.isProduction = config.mode === "production";
             pluginState.enabled =
                 pluginState.sourcemapsCreated &&
-                    (skipEnvironmentCheck || config.isProduction);
+                    (skipEnvironmentCheck || pluginState.isProduction);
         },
         /*
           Resolve id for virtual module